### PR TITLE
COMP: Show local definitions in top of the completion list

### PR DESCRIPTION
**Pull request by mchernyavsky**
Closes https://github.com/intellij-rust/intellij-rust/issues/3585.

### CODE REVIEW

File: src/main/kotlin/org/rust/lang/core/psi/ext/RsElement.kt [hunk 1]
```diff
@@ -145,3 +146,10 @@ fun RsElement.findInScope(name: String): PsiElement? {
     }
     return resolved
 }
+
+
+val RsElement.isLocal: Boolean
+    get() {
+        val context = PsiTreeUtil.getContextOfType(this, true, RsItemElement::class.java, RsFile::class.java)
+        return context != null && context !is RsMod
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/psi/ext/RsElement.kt]: Is it expected that context can be RsMembers and so methods will be classified as local?

Comment (vlad20012) [reply src/main/kotlin/org/rust/lang/core/psi/ext/RsElement.kt]: And enum item too

Comment (mchernyavsky) [reply src/main/kotlin/org/rust/lang/core/psi/ext/RsElement.kt]: > Is it expected that context can be trait or impl and so methods themselves will be classified as locals?

I think yes. Examples from Kotlin:
image

image

File: src/main/kotlin/org/rust/lang/core/psi/ext/RsElement.kt [hunk 2]
```diff
@@ -145,3 +143,10 @@ fun RsElement.findInScope(name: String): PsiElement? {
     }
     return resolved
 }
+
+
+val RsElement.isLocal: Boolean
```
Comment (vlad20012) [inline src/main/kotlin/org/rust/lang/core/psi/ext/RsElement.kt]: Looks like a something very specific to completion sorting. Maybe move it to LookupElements.kt and make private?

**Pull request merged**